apps/web/src/app/next_api/fetch-hsl-stops/route.ts

```diff
@@ -0,0 +1,229 @@
+  } else if (isTram(arrival)) {
+    return "tram";
+  }
+  return "bus";
+}
+function toOutData(stop: Stop | null): StopOutData | null {
+  if (!stop) return null;
+  return {
+    name: stop.name,
+    type: stop.type,
+    arrival: stop.stoptimesWithoutPatterns.map((arr: StopTime) => ({
+      route: arr.trip.routeShortName,
+      headSign: arr.headsign,
+      realTimeArrival: arr.realtimeArrival + arr.serviceDay,
+      serviceDay: arr.serviceDay,
+    })),
+  };
+}
+
+const getData = async (stop: string): Promise<Stop | null> => {
+  const defaultOptions: DefaultOptions = {
+    query: {
+      fetchPolicy: "no-cache",
+    },
+  };
+
+  const client = new ApolloClient<object>({
+    uri: "https://api.digitransit.fi/routing/v1/routers/hsl/index/graphql",
+    cache: new InMemoryCache({
+      resultCaching: false,
+    }),
+    defaultOptions,
+    headers: {
+      "Content-Type": "application/json",
+      "digitransit-subscription-key":
+        process.env.DIGITRANSIT_SUBSCRIPTION_KEY ?? "",
+    },
+  });
+  let data: StopFromApi | null = null;
+
+  await client
+    .query({
+      query: GetStopSchedule(stop),
+    })
+    .then((result: ApolloQueryResult<Data>) => {
+      data = result.data.stop;
+    });
+
+  return data;
+};
+
+function makePrintTime(arrival: Arrival): string {
+  const date = new Date();
+  const hour = date.getHours();
+  const min = date.getMinutes();
+  const sec = date.getSeconds();
+  const currentTSM = (hour * 60 + min) * 60 + sec;
+  let arrivalTime = arrival.realTimeArrival - arrival.serviceDay;
+  if (arrivalTime / 3600 >= 24) {
+    arrivalTime -= 24 * 3600;
+  }
+  if (arrivalTime - currentTSM > 600) {
+    return `${pad(
+      Math.floor((arrival.realTimeArrival - arrival.serviceDay) / 60 / 60) % 24,
+      2,
+    )}:${pad(
+      Math.floor(((arrival.realTimeArrival - arrival.serviceDay) / 60) % 60),
+      2,
+    )}`;
+  }
+  const t1 = Math.floor(arrivalTime - currentTSM);
+  if (t1 < -60) {
+    return "NaN";
+  }
+  const t = Math.floor(Math.max(t1, 0) / 60);
+  return t <= 1 ? "~0" : String(t);
+}
+
+const getStop = async (
+  stops: readonly [string, string],
+  n = N_ARRIVALS,
+): Promise<RenderableStop | null> => {
+  const [result1, result2] = await Promise.all(
+    stops.map((stop) => getData(stop).then(toOutData)),
+  );
+
+  if (!result1 || !result2) return null;
+
+  const result: StopOutData = {
+    name: result1.name,
+    type: getType(result1.arrival),
+    arrival: result1.arrival
+      .map((arr) => arr)
+      .concat(result2.arrival.map((arr) => arr))
+      .sort((arr1, arr2) => arr1.realTimeArrival - arr2.realTimeArrival)
+      .slice(0, n + 1),
+  };
+  const ArrivalsFormatted: ArrivalAttribute[] = result.arrival
+    .map((arr: Arrival) => {
+      return {
+        route: arr.route ? arr.route.replace(" ", "") : "Null",
+        headSign: removeSubstring(arr.headSign),
+        hours:
+          Math.floor((arr.realTimeArrival - arr.serviceDay) / 60 / 60) % 24,
```

Comment:
If the API response times are in any reasonable time format (timestamp or ISO etc.), we could probably use more native datetime operations here, but this mankelointi is probably fine for now.